Makefile

```diff
@@ -3,24 +3,33 @@ all: test build build-so
 out:
 	mkdir -p out
 
-build: out
+internal/connect/version.txt:
+	# this is the equivalent from _service of: @PARENT_TAG@~git@TAG_OFFSET@.%h
+	parent=$$(git describe --tags --abbrev=0 --match='v*' | sed 's:^v::' ); \
+	       offset=$$(git rev-list --count "v$${parent}..HEAD"); \
+	       git log --no-show-signature -n1 --date='format:%Y%m%d' --pretty="format:$${parent}~git$${offset}.%h" > internal/connect/version.txt
+	cat -v internal/connect/version.txt
+	@echo
+
+build: out internal/connect/version.txt
 	go build -v -o out/ github.com/SUSE/connect-ng/suseconnect
 
-test:
+test: internal/connect/version.txt
 	go test -v ./internal/connect
```

Comment:
should the file be removed after? Or should it be in gitignore?

Reply:
Added to .gitignore.